docs/adr/0009_versioning_branching_releasing.md

```diff
@@ -0,0 +1,465 @@
+   * Check if corresponding final release exists (e.g., `ocm/v0.11.0`)
+   * **If no final release exists**: Create new RC with incremented number (rc.1, rc.2, rc.3, ...)
+   * **If final release exists**: Create new RC for next version (v0.12.0-rc.1)
+3. Update component constructor with new sub-component versions
+4. Run integration conformance tests from `/ocm/tests/` (if available)
+5. Manual promotion: RC → Final release (e.g., `ocm/v0.11.0`)
+
+### Failure Recovery for OCM Root Component
+
+**When OCM Root Component RC Conformance Tests Fail**:
+
+1. **Root Cause Analysis**: Identify which sub-component interaction(s) caused the failure
+2. **Sub-component Patch Release**: Create patch releases for affected sub-components on their respective release branches
+3. **Patch Validation**: Each sub-component patch goes through normal testing (hotfix releases bypass RC workflow)
+4. **RC Strategy**: Each sub-component patch release creates a new RC with incremented number (following standard RC numbering)
+5. **Re-test Integration**: Run conformance tests for the new RC with updated sub-component versions
+6. **Iterative Process**: Repeat until conformance tests pass
+
+**Example**:
+
+```text
+ocm/v0.12.0-rc.1 fails → cli/v0.31.1 patch → ocm/v0.12.0-rc.2 with CLI v0.31.1
+```
+
+### OCM Integration Testing
+
+The `/ocm/` directory contains:
+
+* **Component Constructor**: `/ocm/component-constructor.yaml` - defines sub-component versions and relationships
+* **Conformance Tests**: `/ocm/tests/` - integration tests covering scenarios where multiple components interact. These tests validate that the specific combination of sub-component versions works together as expected. They will be created over time and are not mandatory for the initial rollout of this strategy.
+* **Test Scenarios**: End-to-end workflows validating component interoperability beyond individual component tests
+
+These conformance tests are executed during the RC-to-final promotion process to ensure component compatibility.
+
+## CI/CD Workflows and Automation
+
+The release strategy is implemented through several automated workflows that handle different aspects of the release process:
+
+### Workflow: `create-release-branch`
+
+**Trigger**: Manual workflow dispatch OR automated trigger when preparing new minor release
+
+**Purpose**: Initialize a new persistent release branch for a component's minor version line
+
+**Required Steps**:
+
+1. **Branch validation**: Verify that `releases/<component>/v<major>.<minor>` doesn't already exist
+2. **Source determination**: Create branch from `main` for new minor version release
+3. **Branch creation**: Create `releases/<component>/v<major>.<minor>` branch using OCM bot
+4. **Protection rules**: Apply branch protection rules identical to `main` branch (require PR reviews, status checks, admin override settings)
+5. **Notification**: Alert component maintainers of new release branch creation
```

Comment:
Will there be a maintainers file per component or some other structure? This is the first time I'm hearing about individual maintainers / components and notifications based on that.